rest/data/commit_queue.go

```diff
@@ -38,6 +46,268 @@ func (pc *DBCommitQueueConnector) GetGitHubPR(ctx context.Context, owner, repo s
 	return pr, nil
 }
 
+func (pc *DBCommitQueueConnector) AddPatchForPr(ctx context.Context, projectRef model.ProjectRef, prNum int, modules []restModel.APIModule, messageOverride string) (string, error) {
+	settings, err := evergreen.GetConfig()
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get evergreen settings")
+	}
+	githubToken, err := settings.GetGithubOauthToken()
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get github token")
+	}
+	pr, err := thirdparty.GetPullRequest(ctx, prNum, githubToken, projectRef.Owner, projectRef.Repo)
+	if err != nil {
+		return "", err
+	}
+
+	patchDoc, err := patch.MakeNewMergePatch(pr, projectRef.Id, evergreen.CommitQueueAlias, messageOverride)
+	if err != nil {
+		return "", errors.Wrap(err, "unable to make commit queue patch")
+	}
+
+	p, patchSummaries, projectConfig, err := getPatchInfo(ctx, githubToken, patchDoc)
+	if err != nil {
+		return "", err
+	}
+
+	errs := validator.CheckProjectSyntax(projectConfig)
+	errs = append(errs, validator.CheckProjectSettings(projectConfig, &projectRef)...)
+	catcher := grip.NewBasicCatcher()
+	for _, validationErr := range errs.AtLevel(validator.Error) {
+		catcher.Add(validationErr)
+	}
+	if catcher.HasErrors() {
+		update := units.NewGithubStatusUpdateJobForProcessingError(
+			commitqueue.GithubContext,
+			pr.Base.User.GetLogin(),
+			pr.Base.Repo.GetName(),
+			pr.Head.GetRef(),
+			units.InvalidConfig,
+		)
+		update.Run(ctx)
+		grip.Error(message.WrapError(update.Error(), message.Fields{
+			"message":    "error updating pull request with merge error",
+			"project":    projectRef.Identifier,
+			"pr":         prNum,
+			"merge_errs": catcher.Resolve(),
+		}))
+		return "", errors.Wrap(catcher.Resolve(), "errors found validation project configuration file")
+	}
+
+	if err = writePatchInfo(patchDoc, patchSummaries, p); err != nil {
+		return "", err
+	}
+
+	modulePRs, modulePatches, err := getModules(ctx, githubToken, prNum, modules, projectConfig)
+	if err != nil {
+		return "", err
+	}
+	patchDoc.Patches = append(patchDoc.Patches, modulePatches...)
+
+	// populate tasks/variants matching the commitqueue alias
+	projectConfig.BuildProjectTVPairs(patchDoc, patchDoc.Alias)
+
+	if err = addMergeTaskAndVariant(patchDoc, projectConfig, &projectRef, commitqueue.SourcePullRequest); err != nil {
+		return "", err
+	}
+
+	if err = patchDoc.Insert(); err != nil {
+		return "", errors.Wrap(err, "unable to add patch")
+	}
+
+	catcher = grip.NewBasicCatcher()
```

Comment:
we don't really need a new catcher, we know the catcher we already have has no errors so